src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -57,11 +57,19 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
                     var framework_dep = framework?.Dependencies.FirstOrDefault(
                         dep => StringComparer.OrdinalIgnoreCase.Equals(dep.Name, library.Name));
 
+                    CentralPackageVersion centralPackageVersion = null;
+                    var libraryVersionIsCentrallyDefined = framework?.CentralPackageVersions.TryGetValue(library.Name, out centralPackageVersion);
+
                     if (framework_dep != null)
                     {
                         dependency.Type = PackageDependencyType.Direct;
                         dependency.RequestedVersion = framework_dep.LibraryRange.VersionRange;
                     }
+                    else if (libraryVersionIsCentrallyDefined.HasValue && libraryVersionIsCentrallyDefined.Value && centralPackageVersion != null)
+                    {
+                        dependency.Type = PackageDependencyType.CentralTransitive;
```

Comment:
I don't understand why centrally defined libraries are automatically assumed to be transitive. Can't they be regular, direct dependencies??

Reply:
Yes they can. The dgspec already has a list of the direct dependency, so no need to mark them. This metadata will allow to decide when to invalidate the packages.loc.json. 
I added more comments in the code as well for future reference.